src/libmeasurement_kit/common/logger.cpp

```diff
@@ -210,41 +209,25 @@ class DefaultLogger : public Logger, public NonCopyable, public NonMovable {
 
     void emit_event_ex(
             const std::string &key, nlohmann::json &&value) override {
-        if (!value.is_object()) {
```

Comment:
Removed event validation from here. Instead now I register a handler for unhandled events in `engine.cpp` and I make sure in there that events are correct. Removes some code duplication.